### ISSUE TITLE
Baritone doesn't use the correct tool breaking for any block that uses an axe, shovel or hoe 

### ISSUE THREAD

**Issue by Redhawk18** [cabaletta/baritone#4319]
It feels like baritone does not have the tool for mushroom blocks set within the code, because I know it can switch to the proper tool for mining with an axe or shovel.

**Comment by ZacSharp** [comment]
Is this the situation/behavior you mean? (recorded using `tweaker:runClient` on 354b26b1d)

https://github.com/cabaletta/baritone/assets/68165024/ff4c3f6e-053f-4ccb-aeae-b87917362fd2

**Comment by Redhawk18** [comment]
> Is this the situation/behavior you mean? (recorded using `tweaker:runClient` on [354b26b](https://github.com/cabaletta/baritone/commit/354b26b1da30516e35d640f701314b232d842079))
> 2024-04-01.23-54-09.mp4

Very strange, is there a way to get the commit latest commit number for my version of baritone? the version command gives a dev release null. Also I'm using the `litematica` command if that affects anything.

**Comment by Redhawk18** [comment]
I found the problem, if an axe is in the hotbar baritone will use it. Else it will not switch the item to the hotbar to use it. I wonder how many blocks this affects and it has to do with how fast it can break the block.

**Comment by ZacSharp** [comment]
> Else it will not switch the item to the hotbar to use it. I wonder how many blocks this affects and it has to do with how fast it can break the block.

Baritone unconditionally moves a pickaxe to your hotbar and no other tools so this affects anything for which a pickaxe is not the fastest tool. There was a pr to fix this (and much more) but it sadly died (#4037).

**Comment by Redhawk18** [comment]
> > Else it will not switch the item to the hotbar to use it. I wonder how many blocks this affects and it has to do with how fast it can break the block.
> 
> Baritone unconditionally moves a pickaxe to your hotbar and no other tools so this affects anything for which a pickaxe is not the fastest tool. There was a pr to fix this (and much more) but it sadly died (#4037).

why did it die? I dont see any problems, looks like it never got merged.

**Comment by ZacSharp** [comment]
I don't know. It's still a draft so I think Brady (ZeroMemes) had further plans for it, but to know why he didn't finish it you'd have to ask him.

**Comment by Redhawk18** [comment]
whats the best way to reach out to him? I'm sure I could dm him on discord but that is bad practice for open source projects.

**Comment by ZacSharp** [comment]
Discord and pinging him here is the ways I know.

**Comment by Redhawk18** [comment]
@ZeroMemes

**Comment by figloalds** [comment]
Yeah, but it won't use a hoe for gathering carrots when running the farm command

**Comment by ZacSharp** [comment]
@figloalds carrots break instantly without using a tool and there is no tool assigned to them. Why would you expect Baritone to use a hoe to break them?

**Comment by figloalds** [comment]
@ZacSharp because of the fortune enchantment which the wiki states to increase drop rates
```Fully grown carrot crops drop 2 to 5 carrots (3 5⁄7 per crop harvested on average). Yield can be increased using a tool enchanted with [Fortune](https://minecraft.fandom.com/wiki/Fortune), with Fortune III harvesting an average of 5 3⁄7 carrots.```
https://minecraft.fandom.com/wiki/Carrot

**Comment by ZacSharp** [comment]
@figloalds ok, though that's a separate concern from this issue (because considering fortune is not an intended feature). I'd ask you to open a new issue, but there probably is one already.

**Comment by Redhawk18** [comment]
After hours more of testing its any block that uses an axe, shovel or hoe as its tool.

**Comment by ZacSharp** [comment]
Yes, as already stated [here](https://github.com/cabaletta/baritone/issues/4319#issuecomment-2032979820), anything but pickaxes behaves like that.
